cavityDisplayGUI.py

```diff
@@ -85,24 +91,34 @@ def __init__(self, parent=None, args=None):
 
                     severityPV = PV(cavityObject.pvPrefix + SEVERITY_SUFFIX)
                     statusPV = PV(cavityObject.pvPrefix + STATUS_SUFFIX)
+                    descriptionPV = PV(cavityObject.pvPrefix + DESCRIPTION_SUFFIX)
 
-                    # This line is meant to initialize the cavityWidget colors and shapes when first launched
+                    # These lines are meant to initialize the cavityWidget color, shape, and descriptionPV values when first launched
                     self.severityCallback(cavityWidget, severityPV.value)
                     self.statusCallback(cavityWidget, statusPV.value)
+                    cavityWidget.setToolTip('initial fault description')
 
                     # .add_callback is called when severityPV changes value
                     severityPV.add_callback(partial(self.severityCallback, cavityWidget))
 
                     # .add_callback is called when statusPV changes value
                     statusPV.add_callback(partial(self.statusCallback, cavityWidget))
 
-    # Updates shape depending on pv value
+                    # .add_callback is called when descriptionPV
+                    descriptionPV.add_callback(partial(self.descriptionCallback, descriptionPV, cavityWidget))
+
+    # Updatses shape depending on pv value
     def severityCallback(self, cavity_widget, value, **kw):
         self.changeShape(cavity_widget,
                          SHAPE_PARAMETER_DICT[value]
                          if value in SHAPE_PARAMETER_DICT
                          else SHAPE_PARAMETER_DICT[3])
 
+    def descriptionCallback(self, pv_name, cavity_widget: QWidget, **kw):
+        shortFaultDescription = pv_name.get(as_string=True)
```

Comment:
This fix still makes me laugh but I love it

Reply:
I bet there's a more elegant solution! You eluded to the possibility that I might be able to identify the pv name through the callback (as opposed to this very brute force method of passing a PV Object called pv_name). The add_callback can identify a PV's value, so if you know how to call its name, I'm open to changing this

Reply:
Can I get an example of using char_value? I'm blindly trying to change value to char_value in the code and not having much luck. I changed pv_name to pv_object though!